projects/plugins/jetpack/extensions/blocks/sharing-button/sharing-sources-block.php

```diff
@@ -0,0 +1,1397 @@
+<?php // phpcs:ignore WordPress.Files.FileName.InvalidClassFileName
+/**
+ * Define all sharing sources.
+ *
+ * phpcs:disable Generic.Files.OneObjectStructurePerFile.MultipleFound
+ */
+
+namespace Automattic\Jetpack\Extensions\Sharing_Button_Block;
+
+use Automattic\Jetpack\Device_Detection\User_Agent_Info;
+use Jetpack_PostImages;
+
+/**
+ * Base class for sharing sources.
+ * See individual sharing classes below for the implementation of this class.
+ */
+abstract class Sharing_Source_Block {
+	/**
+	 * Button style (icon, icon-text, text, or official).
+	 *
+	 * @var string
+	 */
+	public $button_style;
+
+	/**
+	 * Does the service have an official version.
+	 *
+	 * @var bool
+	 */
+	public $smart;
+
+	/**
+	 * Should the sharing link open in a new tab.
+	 *
+	 * @var bool
+	 */
+	protected $open_link_in_new;
+
+	/**
+	 * Sharing unique ID.
+	 *
+	 * @var int
+	 */
+	protected $id;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param int $id       Sharing source ID.
+	 */
+	public function __construct( $id ) {
+		$this->id = $id;
+		/**
+		 * Filter the way sharing links open.
+		 *
+		 * By default, sharing links open in a new window.
+		 *
+		 * @module sharedaddy
+		 *
+		 * @since 3.4.0
+		 *
+		 * @param bool true Should Sharing links open in a new window. Default to true.
+		 */
+		$this->open_link_in_new = apply_filters( 'jetpack_open_sharing_in_new_window', true );
+	}
+
+	/**
+	 * Get the protocol to use for a sharing service, based on the site settings.
+	 *
+	 * @return string
+	 */
+	public function http() {
+		return is_ssl() ? 'https' : 'http';
+	}
```

Comment:
Maybe this is an opportunity to get rid of some of the legacy things. Here for example maybe we can always use HTTPS at this point?